feat: 관심 채널 등록, 삭제, 수정 API 연동

faber-h opened the pull request:
### ✨ 이슈 번호

- #59 

### 📌 설명

- 관심 채널 관련 기능(등록, 삭제, 정렬)을 API와 연동하고, 관련 컴포넌트와 상태 로직을 구현하여 작성한 Pull Request입니다.

### 📃 작업 사항

- [x] API 연동 axios
  - [x] 관심 채널 API 요청에서 axios 인스턴스를 사용하도록 리팩토링
- [x] 관심 채널 등록/삭제 API 연동
  - [x] 채널 컴포넌트 BlankStarIcon 클릭 시 관심 채널 등록 API 호출
  - [x] 관심 채널 컴포넌트 FilledStarIcon 클릭 시 관심 채널 삭제 API 호출
- [x] 관심 채널 수정 API 연동
  - [x] 관심 채널 Drag & Drop 시 순서 변경 결과를 서버에 반영
- [x] 관심 채널 관련 컴포넌트 로직 정리 
  - [x] 관심 채널 관련 컴포넌트 간 props 키 통일 (channelId, channelName, thumbnail 등)

### 💡 참고 사항

- axios 인스턴스 생성 관련 참고사항
- 기존에는 API 호출 시마다 ${import.meta.env.VITE_API_URL}을 반복적으로 작성하고 있었는데, 이 부분을 axios 인스턴스를 생성해 baseURL로 분리하여 중복을 제거하고 가독성을 높였습니다.
- 변경 전
```
axios.post(`${import.meta.env.VITE_API_URL}/users/${userId}/interest-channels`, { channelId });
```
- 변경 후
```
// apis/axiosInstance.js
import axios from "axios";

const instance = axios.create({
  baseURL: import.meta.env.VITE_API_URL,
});

export default instance;

// apis/interestChannels.js
import axiosInstance from "./instance";

export const createInterestChannel = (userId, channelId) =>
  axiosInstance.post(`/users/${userId}/interest-channels`, { channelId });
```
- 해당 사항을 이전 코드에 반영하여 변경하는 [공통 axiosInstance 적용을 통한 API 호출 일관성 개선](https://github.com/Radio-Premium/RadioPremium-FE/issues/155) issue를 생성하여 리팩토링 진행했습니다.

### 💭 리뷰 사항 반영

### ✅ Pull Request 체크 사항

- [x] 가장 최신 브랜치를 pull 하였습니다.
- [x] base 브랜치명을 확인하였습니다.
- [x] 코드 컨벤션을 모두 확인하였습니다.
- [x] 브랜치명을 확인하였습니다.

faber-h committed:
refactor: 관심 채널 동작 정리 및 컴포넌트 prop 통일

- ChannelListItem, FavoriteChannelListItem 구조 통일 (channelId, channelName, thumbnail)
- 버튼 클릭 시 이벤트 버블링 방지 (`e.stopPropagation()`) 추가
- FavoriteChannelList에 썸네일/채널명 전달되도록 수정

faber-h committed:
feat: 관심 채널 등록/삭제 API 연동

faber-h committed:
feat: 관심 채널 순서 변경 API 연동

- 관심 채널 순서 변경 API 호출하는 updateInterestChannel 함수 추가
- handleChannelDragEnd 유틸 함수로 분리하여 관심 채널 순서 변경 처리

faber-h committed:
refactor: API 함수명을 백엔드와 일치하도록 수정

faber-h committed:
refactor: API 경로 중복 제거를 위해 BASE_URL 상수로 추출

faber-h committed:
refactor: axios 인스턴스 생성 및 관심 채널 API에 적용

- baseURL과 공통 headers가 설정된 axios 인스턴스를 생성
- 관심 채널 관련 API 함수(create, delete, update)에 인스턴스 적용
- 코드 중복 제거

faber-h committed:
rename: 파일명 interestChannels.js로 변경 (API 경로와 통일)

- API 경로(`/users/:userId/interest-channels`)와 명칭을 일치시키기 위해 파일명을 복수형으로 변경

devjaeyoon commented on src/utils/dragHandlers.js:
```diff
@@ -0,0 +1,30 @@
+  userId
+) => {
+  const { active, over } = event;
+  if (!over || active.id === over.id) return;
```
코드 규칙에 따라 중괄호 넣어주면 좋을 것 같아요!

faber-h replied:
감사합니다!
중괄호 감싸서 추가했습니다~

devjaeyoon commented on src/components/FavoriteChannelListItem.jsx:
```diff
@@ -4,8 +4,9 @@ import { CSS } from "@dnd-kit/utilities";
+import useToggleFavorite from "@/hooks/useToggleFavorite";
 
-const FavoriteChannelListItem = ({ id, name }) => {
+const FavoriteChannelListItem = ({ thumbnail, channelId, channelName }) => {
```
저는 DB 테이블에서 id를 최상단에 넣어두는 것처럼 `channelId`가 props의 맨 첫번째로 오는걸 선호하는데 두 분은 어떻게 생각하시는지 궁금해요~ @faber-h @Eun0713

d0-neii replied:
저도 개인적으로는 `channelId`가 주요 식별자라서 앞쪽에 위치하면 의미가 더 잘 드러날 것 같다고 생각해요!
다른 컴포넌트들과도 통일성 있게 이 부분도 `channelId`를 가장 앞에 두자는 재윤님 의견 좋은 것 같습니다 :)

faber-h replied:
좋은 의견 감사해요 ☺️
저도 주요 식별자인 `channelId`를 가장 앞에 두는 게 가독성이나 의미 전달 측면에서 더 명확하다고 생각해요.
반영하여 수정했습니다 :)

faber-h committed:
refactor: 관심 채널 등록, 삭제, 수정 시 유효성 검사 추가

- 관심 채널 등록, 삭제: userId, channelId가 존재하지 않을 경우 API 호출을 막기 위해 early return 추가
- 관심 채널 수정: userId가 존재하지 않을 경우 API 호출을 막기 위해 early return 추가

faber-h committed:
style: 조건문 중괄호 추가

faber-h committed:
refactor: 주요 식별자인 channelId를 props 최상단으로 위치 조정

devjaeyoon reviewed (approved):
LGTM!

d0-neii commented on src/apis/interestChannels.js:
```diff
@@ -0,0 +1,16 @@
+import axiosInstance from "@/apis/axiosInstance";
```
관심 채널 관련 API 요청들을 함수로 잘 분리해주셔서 코드 가독성도 좋고, 재사용성도 높아진 것 같아요!
추후 유지보수에도 유리할 것 같습니다💯 저도 이 스타일 참고해서 앞으로 이렇게 작성해보겠습니다!

faber-h replied:
감사합니다~
해당 내용 참고 사항에 작성해놓았으니 해당 내용도 참고하시면 좋을 것 같아요 😀

Eun0713 reviewed (commented):
LGTM🤩
